src/main/java/org/wildfly/common/function/Functions.java

```diff
@@ -32,6 +33,38 @@
 public final class Functions {
     private Functions() {}
 
+    private static final Logger LOGGER = Logger.getLogger(Functions.class);
+    private static final Function<Object, Object> IDENTITY_FUNCTION = Function.identity();
+
+    /**
+     * Returns a function that always returns its input argument.
+     * A subclass permissive variation of {@link Function#identity()}.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T extends R, R> Function<T, R> identityFunction() {
+        return (Function<T, R>) IDENTITY_FUNCTION;
+    }
```

Comment:
This might be better generalized as something like:

```java
public static <T, R, T2 extends T, R2 super R> Function<T2, R2> loosenBoundsOf(Function<T, R> function) {
    return (Function<T2, R2>) function;
}
```

or something like that.

Reply:
I think you mean:
```java
public static <T, R extends R2, T2 extends T, R2> Function<T2, R2> loosenBoundsOf(Function<T, R> function) {
    return (Function<T2, R2>) function;
}
```
In which case, we should consider a different name.  While this method can loosen the bounds of the return type, it can also constrain the bounds of the argument (or both).
e.g.
```java
Function<Number, Number> function = Function.identity();
Function<Integer, Object> result = loosenBoundsOf(function);
```
Maybe just name it "cast"?
Also, this suggests analogous methods for Supplier, Consumer, BiFunction, exception variants, etc.